src/st/st0/2F774.c

```diff
@@ -0,0 +1,254 @@
+#include "st0.h"
+
+void func_801AF774(Entity* self) {
+    RECT sp10;
+    DRAWENV sp18;
+    DRAWENV* var_a2;
+    DRAWENV* var_a3;
+    Primitive* prim;
+    s32 primIndex;
+    s32 i;
+    u32 var_t0;
+    u8 temp_t0;
+    u8* var_a2_3;
+    s16* temp_a3;
+    s32 xBase;
+    s32 yBase;
+    inline s32 add(a, b) { return a + b; }
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_801805E0);
+        self->hitboxState = 0;
+        self->animCurFrame = 0;
+        self->drawMode |= 0x30;
+        self->flags &= ~FLAG_UNK_08000000;
+        D_801C2578 = 1;
+        self->ext.aguneaCrash.unk8B[5] = 1;
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 0x5C);
+        if (primIndex != -1) {
+            prim = &g_PrimBuf[primIndex];
+            self->primIndex = primIndex;
+            self->ext.et_801AF774.prim3 = prim;
+            self->flags |= FLAG_HAS_PRIMS;
+            prim->tpage = 0x110;
+            if (!self->facingLeft) {
+                prim->u0 = prim->u2 = 0;
+                prim->u1 = prim->u3 = 0x80;
+                prim->v0 = prim->v1 = 0;
+                prim->v2 = prim->v3 = 0x80;
+            } else {
+                prim->u1 = prim->u3 = 0;
+                do {
+                    prim->u0 = prim->u2 = 0x7F;
+                    prim->v0 = prim->v1 = 0;
+                    prim->v2 = prim->v3 = 0x80;
+                } while (0);
+            }
+
+            prim->x0 = prim->x2 = self->posX.i.hi - 0x40;
+            prim->x1 = prim->x3 = self->posX.i.hi + 0x40;
+            prim->y0 = prim->y1 = self->posY.i.hi - 0x25;
+            prim->y2 = prim->y3 = self->posY.i.hi + 0x5B;
+            prim->r0 = prim->g0 = prim->b0 = 0x80;
+            LOW(prim->r1) = LOW(prim->r0);
+            LOW(prim->r2) = LOW(prim->r0);
+            LOW(prim->r3) = LOW(prim->r0);
+            prim->drawMode = 0xA;
```

Comment:
```suggestion
            prim->drawMode = DRAW_UNK02 | DRAW_HIDE;
```
let's start documenting `drawMode`